sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_azure_appconfiguration_client.py

```diff
@@ -422,7 +431,19 @@ def delete_configuration_setting(self, key, label=None, **kwargs):
                 key="MyKey", label="MyLabel"
             )
         """
+        key = kwargs.pop("key", None)
+        label = kwargs.pop("label", None)
         etag = kwargs.pop("etag", None)
+        if len(args) > 0:
+            if isinstance(args[0], ConfigurationSetting):
+                key = args[0].key
+                label = args[0].label
+                if etag is None:
+                    etag = args[0].etag
+            else:
+                key = args[0]
+                if len(args) == 2:
+                    label = args[1]
```

Comment:
else:
TypeError

Reply:
Why TypeError? label is not required, if it is not provided it is defaulted to `None`

Reply:
My bad, I meant if len(args) > 2
TypeError

Reply:
Ahh, yes I will update this in both places.